explorer/syntax/parser.ypp

```diff
@@ -839,57 +902,63 @@ struct_type_literal_contents:
 // specified.
 pattern:
   non_expression_pattern
-    { $$ = $1; }
+    { $$ = $[non_expression_pattern]; }
 | expression
-    { $$ = arena->New<ExpressionPattern>($1); }
+    { $$ = arena->New<ExpressionPattern>($[expression]); }
 ;
 non_expression_pattern:
   AUTO
     { $$ = arena->New<AutoPattern>(context.source_loc()); }
 | binding_lhs COLON pattern
     {
-      $$ = arena->New<BindingPattern>(context.source_loc(), $1, $3,
-                                      std::nullopt);
+      $$ = arena->New<BindingPattern>(context.source_loc(), $[binding_lhs],
+                                      $[pattern], std::nullopt);
     }
 | binding_lhs COLON_BANG expression
     {
-      $$ = arena->New<GenericBinding>(context.source_loc(), $1, $3,
+      $$ = arena->New<GenericBinding>(context.source_loc(), $[binding_lhs],
+                                      $[expression],
                                       GenericBinding::BindingKind::Checked);
     }
 | TEMPLATE binding_lhs COLON_BANG expression
     {
-      $$ = arena->New<GenericBinding>(context.source_loc(), $2, $4,
+      $$ = arena->New<GenericBinding>(context.source_loc(), $[binding_lhs],
+                                      $[expression],
                                       GenericBinding::BindingKind::Template);
     }
 | paren_pattern
-    { $$ = $1; }
+    { $$ = $[paren_pattern]; }
 | postfix_expression tuple_pattern
     {
       ErrorOr<Nonnull<AlternativePattern*>> alternative_pattern =
-          AlternativePattern::Create(arena, context.source_loc(), $1, $2);
+          AlternativePattern::Create(arena, context.source_loc(),
+                                     $[postfix_expression], $[tuple_pattern]);
       if (alternative_pattern.ok()) {
         $$ = *alternative_pattern;
       } else {
         context.RecordSyntaxError(std::move(alternative_pattern).error());
         YYERROR;
       }
     }
-| VAR non_expression_pattern
-    { $$ = arena->New<VarPattern>(context.source_loc(), $2); }
+| VAR non_expression_pattern[var_pattern]
```

Comment:
This feels a little confusing because the most salient "var pattern" here is `$$` itself. Maybe `rhs_pattern` instead?

Reply:
Renamed to just pattern (if it's rhs, what's the lhs?)